connector/src/main/java/org/geysermc/connector/network/translators/java/JavaAdvancementsTranslator.java

```diff
@@ -50,20 +52,30 @@ public void translate(ServerAdvancementsPacket packet, GeyserSession session) {
         // Adds advancements to the player's stored advancements when advancements are sent
         // Also sends notifications for any new advancements
         for (Advancement advancement : packet.getAdvancements()) {
-            if (!session.getStoredAdvancements().isEmpty() && !session.getStoredAdvancements().containsValue(advancement)) {
+            if (!session.getStoredAdvancements().isEmpty() && !session.getStoredAdvancements().containsValue(advancement) && advancement != null && advancement.getDisplayData() != null) {
                 String color = AdvancementsUtils.ADVANCEMENT_FRAME_TYPES_TO_COLOR_CODES.get(advancement.getDisplayData().getFrameType().toString());
                 String advancementName = MessageTranslator.convertMessage(advancement.getDisplayData().getTitle(), session.getLocale());
-
-                SetTitlePacket titlePacket = new SetTitlePacket();
-                titlePacket.setText(color + "[" + LocaleUtils.getLocaleString("advancements.toast." + advancement.getDisplayData().getFrameType().toString().toLowerCase(), session.getLocale()) + "] " + advancementName);
-                titlePacket.setType(SetTitlePacket.Type.ACTIONBAR);
-                titlePacket.setFadeOutTime(3);
-                titlePacket.setFadeInTime(3);
-                titlePacket.setStayTime(3);
-
-                session.sendUpstreamPacket(titlePacket);
+                boolean earned = true;
+                if (session.getStoredAdvancementProgress().get(advancement.getId()) != null || session.getStoredAdvancementProgress() != null || !session.getStoredAdvancementProgress().get(advancement.getId()).entrySet().isEmpty()) {
```

Comment:
You should use `packet.getProgress()` instead of `session.getStoredAdvancementProgress()` here